### DISCUSSION TITLE
Progress event should tell model, environment, and skybox loading states apart

### DISCUSSION THREAD

**Discussion by designbyadrian** [google/model-viewer#3629] [Ideas]
### Description

When setting `src` and `environment-image` _and/or_ `skybox-image`, it doesn't seem to be possible to know when which of these are finished loading.

The minor issue is that the progress bar can go backwards when the first of the three is finished, but there's time left for another.

A major issue is that I don't know when I can set cameras and materials when progress is 1.0, because maybe the model isn't done the first time progress is 1.0.

### Version
- model-viewer: v1.10.0

### Browser Affected
All

### OS
All

**Comment by designbyadrian** [comment]
Just to confirm;

does the `load` event fire only when the model is done?

**Comment by samaneh-kazemi** [comment]
Hi @designbyadrian, thanks for reaching out. Can you please share more context and share how you use our API? If you can share a link to your project so we can repro it'd be helpful. And yes, `load` is only triggered when the model loaded.

**Comment by designbyadrian** [comment]
@samaneh-kazemi Alright, I'll make some kind of example, as currently I'm displaying confidential client models :D
